src/lib/remote-place-cache.ts

```diff
@@ -72,17 +93,16 @@ export default class RemotePlaceCache {
     : Promise<RemotePlace[]> {
     const { domain } = chtApi.chtSession.authInfo;
     const placeType = hierarchyLevel.contact_type;
-    const { cache: domainCache } = RemotePlaceCache;
-    const places = domainCache[domain]?.[placeType];
-    if (!places) {
+    const cacheKey = this.getCacheKey(domain, placeType);
+    
+    const cacheData = this.getCache().get<RemotePlace[]>(cacheKey);
+    if (!cacheData) {
       const fetchPlacesWithType = this.fetchRemotePlacesAtLevel(chtApi, hierarchyLevel);
```

Comment:
The reason this was split before is to cover the scenario:  

1. Invoke `fetchCachedOrRemotePlaces`
2. Before the above call completes, call `fetchRemotePlacesAtLevel` again.

Expect: Both calls should not invoke `fetchRemotePlacesAtLevel` because it is very heavy and slow. The second call should be blocked on the result of the first.

I think that behavior has regressed. Can you add a test for it and fix?